src/devices/openxrheadset/PosePublisher.cpp

```diff
@@ -11,6 +11,53 @@
 #include <OpenXrYarpUtilities.h>
 #include <yarp/os/LogStream.h>
 
+bool PosePublisher::positionJumped()
+{
+    if (!m_lastValidData.pose.positionValid || !m_data.pose.positionValid)
+    {
+        return false; //Also the last "valid" pose is not valid, or the input is already not valid
+    }
+
+    if (!m_lastValidData.velocity.linearValid)
+    {
+        yCWarning(OPENXRHEADSET) << "The last valid pose of" << m_label << "has non-valid linear velocity. Assuming there are jumps.";
+        return true;
+    }
+
+    Eigen::Vector3f expectedPosition = m_lastValidData.pose.position + m_settings->period * m_lastValidData.velocity.linear;
+
+    return (expectedPosition - m_data.pose.position).norm() > m_settings->checks.maxDistance;
+}
+
+bool PosePublisher::rotationJumped()
+{
+    if (!m_lastValidData.pose.rotationValid || !m_data.pose.rotationValid)
+    {
+        return false; //Also the last "valid" pose is not valid, or the input is already not valid
+    }
+
+    if (!m_lastValidData.velocity.angularValid)
```

Comment:
I am a bit lost, who is setting these flags?

Reply:
It is ultimately provided by ``OpenXR`` itself: https://github.com/ami-iit/yarp-device-openxrheadset/blob/4ea162bc2e35ccedb795aa268ce67ed60e95fd8d/src/devices/openxrheadset/impl/OpenXrInterfaceImpl.cpp#L140-L156

See https://www.khronos.org/registry/OpenXR/specs/1.0/html/xrspec.html#xrLocateSpace

Reply:
Thanks!